src/SourceLocator/Type/MemoizingSourceLocator.php

```diff
@@ -0,0 +1,83 @@
+<?php
+declare(strict_types=1);
+
+namespace Roave\BetterReflection\SourceLocator\Type;
+
+use Roave\BetterReflection\Identifier\Identifier;
+use Roave\BetterReflection\Identifier\IdentifierType;
+use Roave\BetterReflection\Reflection\Reflection;
+use Roave\BetterReflection\Reflector\Reflector;
+
+final class MemoizingSourceLocator implements SourceLocator
```

Comment:
TODO: test this.

Also, this makes no huge difference on the test suite itself (the suite manually instantiates locators most of the time)

Reply:
Tested